authzed/api/v1/experimental_service.proto

```diff
@@ -41,6 +43,75 @@ service ExperimentalService {
         body: "*"
       };
     }
+
+
+  rpc StreamingBulkCheckPermission(StreamingBulkCheckPermissionRequest)
+      returns (stream StreamingBulkCheckPermissionResponse) {
+    option (google.api.http) = {
+      post: "/v1/experimental/permissions/streamingbulkcheckpermission"
+      body: "*"
+    };
+  }
+
+  rpc BulkCheckPermission(BulkCheckPermissionRequest)
+      returns (BulkCheckPermissionResponse) {
+    option (google.api.http) = {
+      post: "/v1/experimental/permissions/bulkcheckpermission"
+      body: "*"
+    };
+  }
+}
+
+message StreamingBulkCheckPermissionRequest {
+  Consistency consistency = 1;
+
+  repeated BulkCheckPermissionRequestItem items = 2 [ (validate.rules).repeated .items.message.required = true ];
+}
+
+message BulkCheckPermissionRequest {
+  Consistency consistency = 1;
+
+  repeated BulkCheckPermissionRequestItem items = 2 [ (validate.rules).repeated .items.message.required = true ];
+}
+
+message BulkCheckPermissionRequestItem {
+  ObjectReference resource = 1 [ (validate.rules).message.required = true ];
```

Comment:
Was there any thought given to having these components be exploded/repeatable as well?

Then I could do things like:
`check(document:1#read@user:(1,2,3,4,5))`
or 
`check(document:(1,2,3,4,5)#read@user:1`

as part of a larger bulk check?

Basically I assume that a lot of bulk-checking use-cases would be bulk checking with most parameters fixed, and just one being "bulk". This API requires that you repeat yourself a lot.

Reply:
Yeah, I think that this API fits the simplest possible client logic (e.g. `resources.map(x => x.proto())`). It definitely uses more bandwidth, but I think it's a worthy tradeoff for UX.

I'd love to see how it plays out as an experiment and we can change it when if necessary.